src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -145,8 +145,35 @@ private AnalyzerConfigSet(ImmutableArray<AnalyzerConfig> analyzerConfigs, Global
             Debug.Assert(allMatchers.Count == _analyzerConfigs.Length);
 
             _analyzerMatchers = allMatchers.ToImmutableAndFree();
+
+            // parse the global options upfront
```

Comment:
what's the motivation to move this parsing of global options up? (why not keep it in `GetOptionsForSourcePath`?)

Reply:
Because we're returning them both as a property and inline in `GetOptionsForSourcePath` we should probably just make it lazy though, rather than doing it unconditionally.

---
In reply to: [471683120](https://github.com/dotnet/roslyn/pull/45871#discussion_r471683120) [](ancestors = 471683120)